rust/hyperlane-base/src/server/eigen_node.rs

```diff
@@ -0,0 +1,303 @@
+pub struct EigenNodeAPI {
+    core_metrics: Arc<CoreMetrics>,
+}
+
+impl EigenNodeAPI {
+    /// Function to create the eigen_node_router
+    pub fn router(&self) -> Router {
+        // let health_api_clone = self.clone();
+        let core_metrics_clone = self.core_metrics.clone();
+
+        Router::new()
+            .route(
+                "/node/health",
+                get(move || Self::node_health_handler(core_metrics_clone)),
+            )
+            .route("/node/services", get(Self::node_services_handler))
+            .route(
+                "/node/services/:service_id/health",
+                get(Self::service_health_handler),
+            )
+            .route("/node", get(Self::node_info_handler))
+    }
+
+    /// Method to return the NodeInfo data
+    pub async fn node_info_handler() -> impl IntoResponse {
+        let node_info = NodeInfo {
+            node_name: "Hyperlane Validator".to_string(),
+            spec_version: "0.1.0".to_string(),
+            node_version: "0.1.0".to_string(),
+        };
+        Json(node_info)
+    }
+
+    /// Method to return the NodeInfo data
+    /// if signed_checkpoint - observed_checkpoint <= 1 return 200 - healthy
+    /// else if observed_checkpoint - signed_checkpoint <= 10 return 203 - partially healthy
+    /// else return 503 - unhealthy
+    pub async fn node_health_handler(core_metrics: Arc<CoreMetrics>) -> impl IntoResponse {
+        let observed_checkpoint = core_metrics
+            .latest_checkpoint()
+            .with_label_values(&["validator_observed", "ethereum"])
+            .get();
+        println!("latest_checkpoint: {}", observed_checkpoint);
+        let signed_checkpoint = core_metrics
+            .latest_checkpoint()
+            .with_label_values(&["validator_processed", "ethereum"])
+            .get();
+
+        // logic to check if the node is healthy
+        if observed_checkpoint - signed_checkpoint <= 1 {
+            // 200 - healthy
+            StatusCode::OK
+        } else if observed_checkpoint - signed_checkpoint <= 10 {
+            // 206 - partially healthy
+            StatusCode::PARTIAL_CONTENT
+        } else {
+            // 503 - unhealthy
+            StatusCode::SERVICE_UNAVAILABLE
+        }
+    }
+
+    /// Method to return a list of services
+    /// NB: hardcoded for now
+    pub async fn node_services_handler() -> impl IntoResponse {
+        let services = vec![
+            Service {
+                id: "hyperlane-validator-indexer".to_string(),
+                name: "indexer".to_string(),
+                description: "indexes the messages from the origin chain mailbox".to_string(),
+                status: ServiceStatus::Up,
+            },
+            Service {
+                id: "hyperlane-validator-submitter".to_string(),
+                name: "submitter".to_string(),
+                description: "signs messages indexed from the indexer".to_string(),
+                status: ServiceStatus::Down,
```

Comment:
down?

Reply:
it was just an example but fixed

Reply:
issue here: https://github.com/hyperlane-xyz/issues/issues/1070